ext/mbstring/libmbfl/filters/mbfilter_ucs2.c

```diff
@@ -216,3 +227,110 @@ static int mbfl_filt_conv_ucs2_wchar_flush(mbfl_convert_filter *filter)
 
 	return 0;
 }
+
+static size_t mb_ucs2_to_wchar(unsigned char **in, size_t *in_len, uint32_t *buf, size_t bufsize, unsigned int *state)
```

Comment:
Won't this try to determine endianness again once the output buffer has been exhausted and the filter is called again?

I think you should be using the state here with 0 unknown endianness and 1/2 little/big endian either explicit or implicit.

Reply:
Also applies to all the other encodings doing endianness detection...

Reply:
Very, very good point. I'll fix that.

Reply:
Done.